lib/internal/fs/linux_recursive_watcher.js

```diff
@@ -0,0 +1,154 @@
+'use strict';
+
+const { EventEmitter } = require('events');
+const path = require('path');
+const { Symbol, ObjectKeys } = primordials;
+
+const kFSWatchStart = Symbol('kFSWatchStart');
+
+let internalSync;
+let internalPromises;
+
+function lazyLoadFsPromises() {
+  internalPromises ??= require('fs/promises');
+  return internalPromises;
+}
+
+function lazyLoadFsSync() {
+  internalSync ??= require('fs');
+  return internalSync;
+}
+
+async function traverse(dir, files = {}) {
+  const { stat, readdir } = lazyLoadFsPromises();
+
+  files[dir] = await stat(dir);
+
+  try {
+    const directoryFiles = await readdir(dir);
+
+    for (const file of directoryFiles) {
+      const f = path.join(dir, file);
+
+      try {
+        const stats = await stat(f);
+
+        files[f] = stats;
+
+        if (stats.isDirectory()) {
+          await traverse(f, files);
+        }
+      } catch (error) {
+        if (error.code !== 'ENOENT' || error.code !== 'EPERM') {
+          throw error;
+        }
+      }
+
+    }
+  } catch (error) {
+    if (error.code !== 'EACCES') {
+      throw error;
+    }
+  }
+
+  return files;
+}
+
+class FSWatcher extends EventEmitter {
+  #options = null;
+  #closed = false;
+  #files = {};
+
+  /**
+   * @param {{
+   *   persistent?: boolean;
+   *   recursive?: boolean;
+   *   encoding?: string;
+   *   signal?: AbortSignal;
+   *   }} [options]
+   */
+  constructor(options) {
```

Comment:
```suggestion
  constructor(options = kEmptyObject) {
```

Reply:
Thanks. I just updated the pull request.